### DISCUSSION TITLE
Built-in JS Minify Builder (With sourcemaps)

### DISCUSSION THREAD

**Discussion by trymeouteh** [rollup/rollup#5506] [Ideas]
Rollup does not support minifying JS code out of the box and require extensions to do so. I am unable to get these extensions to work since they are not updated to work for the latest version of Rollup.

It would be handy to have a minifier built into Rollup which also supports sourcemaps for easy debugging!

**Comment by lukastaegert** [comment]
> I am unable to get these extensions to work since they are not updated to work for the latest version of Rollup.

I would recommend this one which IS updated to work with Rollup 4: https://www.npmjs.com/package/@rollup/plugin-terser

That being said, I would not completely ignore the idea as minifying in Rollup would be quite a bit faster than running terser after it. But it would be a huge undertaking and it will very likely not happen this year.

**Reply by trymeouteh** [reply]
> minifying in Rollup would be quite a bit faster than running terser after it. But it would be a huge undertaking and it will very likely not happen this year.

Agreed and understood
